bundles/org.openhab.binding.amazonechocontrol/src/main/java/org/openhab/binding/amazonechocontrol/internal/Connection.java

```diff
@@ -322,7 +342,7 @@ public boolean tryRestoreLogin(@Nullable String data, @Nullable String overloade
         Scanner scanner = new Scanner(data);
         String version = scanner.nextLine();
         // check if serialize version
-        if (!version.equals("5") && !version.equals("6")) {
+        if (!version.equals("5") && !version.equals("6") && !version.equals("7")) {
```

Comment:
Just a general remark, which I didn't notice before.

Position literals first in `String` comparisons - that way if the `String` is `null` you won't get a `NullPointerException`, it'll just return `false`. [source: PMD](http://pmd.sourceforge.net/pmd-4.3.0/rules/design.html#PositionLiteralsFirstInComparisons)

Reply:
Changed